### PR TITLE
feat: change from sync to async work

### PR DESCRIPTION
**Pull request by asiman161**
Hello! 
I started to look Binance rust clients and only this was actual. I saw some issues that users don't want to use this lib because it's sync. (as me. I forked it to fix this problem. I don't want to use workaround. Write my own lib or fork this and fix)
Today I don't know why it should work with sync only (historical? then I want to have async version) and I thought that it will be useful to move it to async API. 

I checked examples and tests and it works.

### CODE REVIEW

**Comment by danieleades** [comment]
@asiman161 i'm afraid i made a couple of PRs that may mean you need to rebase.

I hope this PR gets merged soon! I think 'async-first' is the right approach for API clients

**Comment by Sydney-o9** [comment]
@asiman161 Are you able to update it for @danieleades to merge? Alternatively, let me know, I am currently using a rebased fork which brings your changes - see:
https://github.com/NextonCapital/binance-rs/tree/feature/async-binance-rs

**Comment by Sydney-o9** [comment]
@danieleades @wisespace-io @asiman161 No answer from you guys, before you merge new things to master - I've put the PR up so it is not delayed further for your consideration.

https://github.com/wisespace-io/binance-rs/pull/179

**Comment by asiman161** [comment]
@Sydney-o9 Hello! I just read your message :) 
I can update my MR in 1-2 days

**Comment by asiman161** [comment]
@danieleades I updated PR to actual repo version so you can check it

File: Cargo.toml
Comment (danieleades) [inline Cargo.toml:32]: is this a library dependency, or should be a dev depedency only?

File: Cargo.toml
Comment (danieleades) [inline Cargo.toml:29]: do you still need the 'blocking' feature?

**Review by danieleades — CHANGES_REQUESTED** [review]

**Comment by danieleades** [comment]
you'll need @wisespace-io to trigger the workflows for you

File: Cargo.toml
Comment (asiman161) [reply Cargo.toml:29]: no, only tests use it and provided from someone year ago. I think that we can delete it safely (and remove from tests)

Comment (asiman161) [reply Cargo.toml]: I deleted this feature and fixed test

File: Cargo.toml
Comment (asiman161) [reply Cargo.toml]: ok, you're right. fixed it

**Review by danieleades — APPROVED** [review]
LGTM, once the workflows pass. I suspect this will need a `rustfmt` pass before the workflows are happy.

**Comment by asiman161** [comment]
> I suspect this will need a rustfmt pass before the workflows are happy.

months ago this PR has passed workflow but I updated this PR with cargo fmt to avoid any extra problems

**Comment by asiman161** [comment]
@testforvln99 I can't find async crates that's not outdated. 
Async is future and more simpler to write fast apps. The only problem with async that it requires runtime... But it's almost free. 

The only 1 reason that I agree with you is this PR is dangerous because it breaks back compatibility and forces you to use async.

I want to get approve for this PR from @danieleades or get rejected, but we have no answer about it

**Comment by danieleades** [comment]
> I want to get approve for this PR from @danieleades or get rejected, but we have no answer about it

I don't have any authority to approve or reject a PR in this repo

**Comment by asiman161** [comment]
OK. If you speak that this project will be always sync then I should close this PR because it will never be merged.

**Pull request closed without merging**
